overlord/snapstate/handlers.go

```diff
@@ -127,48 +149,47 @@ func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
 	st.Lock()
 	defer st.Unlock()
 
-	// check if core is installed already
-	_, err := CoreInfo(st)
-	if err == nil {
-		return nil
-	}
-	if err != nil && err != state.ErrNoState {
-		return err
-	}
-
 	// check if we need to inject tasks to install core
 	snapsup, _, err := snapSetupAndState(t)
 	if err != nil {
 		return err
 	}
+
+	// core/ubuntu-core can not have prerequisites
 	snapName := snapsup.Name()
 	if snapName == defaultCoreSnapName || snapName == "ubuntu-core" {
```

Comment:
will/can base snap have bases?  didn't we discuss in London that core in the end will also have a base?

Reply:
I would like to avoid this, in my mental model core is self contained. I'm open for discussions here of course.

Reply:
We did discuss that core might need a base, given the needs of hooks/etc, but I agree it would be best to avoid it as it will make things more complex and we've had a lot of pain in the past with general logic stumbling to handle the core case correctly.

I think we should support that as a general feature, actually: snaps may explicitly choose to have no bases, in which case snapd will setup a minimum environment for the snap to run. This will most likely make sense in other situations too (thinking of content providers, etc).